Source/localtracer/include/localtracer/localtracer.h

```diff
@@ -0,0 +1,254 @@
+#pragma once
+
+#include <core/core.h>
+#include <messaging/messaging.h>
+
+#include <iomanip>
+#include <iostream>
+
+namespace WPEFramework {
+namespace Messaging {
+    class EXTERNAL LocalTracer {
+    public:
+        struct EXTERNAL ICallback {
+            virtual void Output(const Core::Messaging::Information& info, const Core::Messaging::IEvent* message) = 0;
+        };
+
+    private:
+        std::string DispatcherIdentifier()
+        {
+            string result;
+            Core::SystemInfo::GetEnvironment(Core::Messaging::MessageUnit::MESSAGE_DISPACTHER_IDENTIFIER_ENV, result);
+            return result;
+        }
+
+        std::string DispatcherBasePath()
+        {
+            string result;
+            Core::SystemInfo::GetEnvironment(Core::Messaging::MessageUnit::MESSAGE_DISPATCHER_PATH_ENV, result);
+            return result;
+        }
+
+    private:
+        class WorkerThread : private Core::Thread {
+        public:
+            WorkerThread(LocalTracer& parent)
+                : Core::Thread()
+                , _parent(parent)
+            {
+            }
+            void Start()
+            {
+                Thread::Run();
+            }
+            void Stop()
+            {
+                Thread::Block();
+            }
+
+        private:
+            uint32_t Worker() override
+            {
+                if (Thread::IsRunning()) {
+                    _parent.Dispatch();
+                }
+
+                return Core::infinite;
+            }
+
+            LocalTracer& _parent;
+        };
+
+    public:
+        LocalTracer(const LocalTracer&) = delete;
+        LocalTracer& operator=(const LocalTracer&) = delete;
+
+        ~LocalTracer()
+        {
+            _worker.Stop();
+
+            _client.SkipWaiting();
+            _client.ClearInstances();
+        }
+
+        void Close()
+        {
+            Core::Messaging::MessageUnit::Instance().Close();
+
+            Core::Directory(_path.c_str()).Destroy();
+
+            delete this;
```

Comment:
You also want to clear the singleton pointer, so that if you open it again (after closing) that it works. So suggest to:

LocalTracer*& singleton = LockSingleton(true);
singleton = nullptr;
LockSingleton(false);

before the delete this...

Reply:
Done